### DISCUSSION TITLE
branch 생성

### DISCUSSION THREAD

**Discussion by ghost** [HeoJiye/Github-Study#6] [질문 게시판] [answered]
branch 생성을 하면
fatal: not a valid object name: 'master'

자꾸 이런 오류가 뜹니다..

**Comment by HeoJiye** [comment, marked as answer]
질문이 올라온 것을 이제 확인했네요!

먼저 git clone을 사용해서 Github-Study 레포지터리를 로컬에 가져오신 게 맞으실까요? Github-Study의 기본 브랜치는 `main`인데, 오류에서는 `master` 브랜치를 사용하고 계신 것 같네요!

새롭게 git clone 다시 시도해보시고, 또 오류가 뜨면 다음 명령어로 브랜치 목록 띄워서 이전 기록들이랑 같이 캡처해서 올려주세요!
```
git branch
```

**Reply by HeoJiye** [reply]
참고로 해당 오류는 clone 없이 로컬에서 레포지터리 생성 후, 최초 커밋 없이 새로운 브랜치를 생성하려 시도했을 때 발생하는 오류인 것 같아요.

다음과 같이 최초 커밋 1번 시도하고 브랜치 생성하면 오류는 해결됩니다.
```
git commit -m "initial commit"
```
<br>

**하지만 깃허브 스터디 1주차 과제는 로컬에서 레포지터리를 생성하기가 아니라 깃허브의 레포지터리를 clone해서 하는 것으로 진행되므로, 1주차 과제에서는 처음 말씀드린 것처럼 git clone을 다시 시도해보세요**

**Reply by ghost** [reply]
지우고 처음부터 다시 시도했는데 해결했습니다 :) 감사합니다!!
